### DISCUSSION TITLE
Fleet problem fresh distributed install - web return 404 error

### DISCUSSION THREAD

**Discussion by p1kusmie** [Security-Onion-Solutions/securityonion#4259] [Unsupported Versions] [answered]
version:  Security Onion 2.3.50
Install source: network install, Ubuntu 18.4
Install type: distributed, problem on ManagerSearch
Installation using: fqdn

There were problems with the fleet after installation. The fleet container works for a while and then goes into error:

![image](https://user-images.githubusercontent.com/47448475/119126889-eac7e780-ba33-11eb-9c42-f5617f1502bc.png)

After executing the command so-fleet-setup user password. The installation stops for a longer time in place:

> [INFO    ] Completed state [/opt/so/conf/so-status/so-status.conf] at time 08:38:46.094473 (duration_in_ms=10.459)
[+] Set the address config key to "https://127.0.0.1:8080" in the "default" context
[+] Set the tls-skip-verify config key to "true" in the "default" context
[+] Set the url-prefix config key to "/fleet" in the "default" context
Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
Updating the Enroll Secret...
[INFO    ] Loading fresh modules for state activity
[INFO    ] Executing command 'docker' in directory '/root'
[ERROR   ] Command 'docker' failed with return code: 1
[ERROR   ] stdout: Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
[ERROR   ] retcode: 1
[ERROR   ] Command 'docker' failed with return code: 1
[ERROR   ] output: Error response from daemon: Container e7021255224dc8da8f26dd6ca5c5a0ccbee5399d2600a5d464cb7edaa0327a81 is not running
[CRITICAL] Rendering SLS 'base:fleet.event_update-enroll-secret' failed: mapping values are not allowed in this context

/opt/so/saltstack/local/pillar/global.sls

> global:
  soversion: 2.3.50
  hnmanager: '10.0.0.0/8,192.168.0.0/16,172.16.0.0/'
  dockernet: '10.10.10.0'
  mdengine: 'ZEEK'
  ids: 'Suricata'
  url_base: 'hostname.xyz.local'
  managerip: 'ip manager'
  airgap: False
  hivekey: 'pass'
  hiveplaysecret: 'pass'
  cortexkey: 'pass'
  cortexorgname: 'SecurityOnion'
  cortexorguserkey: 'pass'
  cortexplaysecret: 'pass'
  fleet_custom_hostname: 'hostname.xyz.local'
  fleet_manager: True
  fleet_node: False
  fleet_packages-timestamp: '2021-05-21-08:42'
  fleet_packages-version: 13
  fleet_hostname: 'hostname'
  fleet_ip: 'manager_ip'
  sensoronikey: 'pass'
  wazuh: 1
  imagerepo: 'security-onion-solutions'
  pipeline: 'redis'
sensoroni:
  node_checkin_interval_ms: 10000
strelka:
  enabled: 1
  rules: 1
  repos:
    - 'https://github.com/Neo23x0/signature-base'
curator:
  hot_warm: False
elastic:
  features: False
elasticsearch:
  replicas: 0
  true_cluster: False
  true_cluster_name: 'so'
  discovery_nodes: 1
  hot_warm_enabled: False
  cluster_routing_allocation_disk.threshold_enabled: true
  cluster_routing_allocation_disk_watermark_low: '95%'
  cluster_routing_allocation_disk_watermark_high: '98%'
  cluster_routing_allocation_disk_watermark_flood_stage: '98%'
  index_settings:
    so-beats:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-firewall:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-flow:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-ids:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-import:
      shards: 1
      warm: 7
      close: 73000
      delete: 73001
    so-osquery:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-ossec:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-strelka:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-syslog:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-zeek:
      shards: 5
      warm: 7
      close: 45
      delete: 365
minio:
  access_key: ''
  access_secret: ''
s3_settings:
  size_file: 2048
  time_file: 1
  upload_queue_size: 4
  encoding: 'gzip'
  interval: 5
backup:
  locations:
    - /opt/so/saltstack/local
soctopus:
  playbook:
    rulesets:
      - windows
docker:
  range: '10.10.10.0/24'
  bip: '10.10.10.1/24'
redis_settings:
  redis_maxmemory: 812
playbook:
  api_key: pass

Thank for help

**Comment by p1kusmie** [comment]
root@onion1m:~# docker attach so-fleet 
Error creating db connection: Error 1045: Access denied for user 'fleetdbuser'@'10.10.10.1' (using password: YES)
Is that a problem with mysql? 
Directory /opt/so/log/fleet/ is empty.

**Comment by defensivedepth** [comment]
It looks like the default Docker IP range was not used, correct?

**Reply by p1kusmie** [reply]
Yes that's true the default network is in conflict with the internal network.

**Reply by defensivedepth** [reply]
I am still trying to replicate this error, but am not able to as of yet.

**Comment by p1kusmie** [comment, marked as answer]
If I reinstall the manager, can I reconnect the remaining nodes without reinstalling to the new one?

**Reply by defensivedepth** [reply]
You would have to reinstall the other nodes. 

What exactly are you trying to fix with reinstalling the manager? The Fleet issue?

**Reply by p1kusmie** [reply]
I have successfully resolved the problem. I added mysql user
CREATE USER 'fleetdbuser'@'10.10.10.1' IDENTIFIED BY 'password';
GRANT USAGE ON *. * TO 'fleetdbuser'@'10.10.10.1';
GRANT ALL PRIVILEGES ON `fleet`. * TO 'fleetdbuser'@'10.10.10.1' 

I don't know why the user 'fleetdbuser@'10.10.10.0/255.255.0.0' did not work even though the log shows requests from the correct address:
2021-06-02T12: 08: 31.167568Z 67578 [Note] Access denied for user 'fleetdbuser'@'10.10.10.1' (using password: YES)

**Reply by dgaudettesn** [reply]
@piotr3439 I've got the exact same issue but can't find how to run mysql  - where can I find the mysql binary to run to add the user?

**Reply by defensivedepth** [reply]
@dgaudettesn Please open up a new Discussion, thanks!
